### PR TITLE
feat(NumberInput): Migrate to styled-components

### LINKED ISSUE #1526: Convert NumberInput to use Styled Components

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
Closes #1526 

## Overview
Migrates the `NumberInput` component to use styled-components.

## Reason
We are migrating all components.

## Work carried out
- [x] Migrate styles
- [x] Migrate `is-valid` and `is-invalid`

### CODE REVIEW

File: packages/react-component-library/src/components/NumberInput/EndAdornment.tsx [hunk 1]
```diff
@@ -35,7 +47,7 @@ export const EndAdornment: React.FC<EndAdornmentProps> = ({
   }
 
   return (
-    <div className="rn-numberinput__controls">
+    <StyledNumberInputControls className="rn-numberinput__controls">
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/NumberInput/EndAdornment.tsx]: Is that className now redundant?

File: packages/react-component-library/src/components/NumberInput/EndAdornmentButton.tsx [hunk 2]
```diff
@@ -1,16 +1,90 @@
...
+const StyledDecreaseButton = styled(StyledButton)`
+  border-top: 1px solid ${color('neutral', '100')};
+
+  &:focus {
+    border-color: transparent;
+  }
+
+  ${({ isCondensed }) =>
+    isCondensed &&
+    css`
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/NumberInput/EndAdornmentButton.tsx:59]: Hmm i've been doing this without the `css`. Is this optional?

File: packages/react-component-library/src/components/NumberInput/NumberInput.tsx [hunk 3]
```diff
@@ -2,13 +2,14 @@ import React from 'react'
 import classNames from 'classnames'
 import { isFinite, isNil } from 'lodash'
 import { selectors } from '@royalnavy/design-tokens'
-import styled, { css } from 'styled-components'
+import styled  from 'styled-components'
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/NumberInput/NumberInput.tsx]: Little bit of extra whitespace here.

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
LGTM just a few tiny comments

File: packages/react-component-library/src/components/NumberInput/EndAdornmentButton.tsx (hunk 2 above)
Comment (thyhjwb6) [reply packages/react-component-library/src/components/NumberInput/EndAdornmentButton.tsx:59]: It seems like it is optional but my IDE does not like it and the documentation I see always seems to use `css`.

With `css`:
Screenshot 2020-10-19 at 13 56 08

Without `css`:
Screenshot 2020-10-19 at 13 56 28

**Review by m7kvqbe1 — APPROVED** [review]
👍

**Pull request merged**